Foundation/NSFileManager.swift

```diff
@@ -340,17 +340,15 @@ public class NSFileManager : NSObject {
         result[NSFileSystemFileNumber] = NSNumber(unsignedLongLong: UInt64(s.st_ino))
         
         let pwd = getpwuid(s.st_uid)
-        if pwd != nil && pwd.memory.pw_name != nil {
-            if let name = String.fromCString(pwd.memory.pw_name) {
-                result[NSFileOwnerAccountName] = name
-            }
+        if pwd != nil && pwd.pointee.pw_name != nil {
+            let name = String(cString: pwd.pointee.pw_name)
+            result[NSFileOwnerAccountName] = name
```

Comment:
In general we're trying to avoid static factory methods and replace them with initializers. This particular initializer (`init(cString:)`) will try to repair ill-formed code units, which, we believe, is the behavior one would prefer by default (as opposed to the failing one). If you do prefer it to fail -- there is `String.init?(validatingUTF8:)`.
Please note though, that both initializers will trap on nil.